### PR TITLE
Upgrade requirements

### PR DESCRIPTION
**Pull request by supertanuki**

### CODE REVIEW

File: tests/Requirement/RequirementCheckerTest.php [hunk 1]
```diff
@@ -135,39 +135,49 @@ public function testCheckerShouldNotCreateAViolationIfBinaryIsOK()
...
 
+        // Version 1.8.4 should generate a violation
+        $violationList = new RequirementViolationList();
+        $this->processor->process('vagrant')->willReturn('vagrant 1.8.4');
+        $this->handlerFactory->getVersionParser()->willReturn(new BinaryVersionParser());
+        $this->requirementChecker->check($requirement, $violationList);
+        var_dump($violationList);
+        $this->assertCount(1, $violationList);
+
         // Version 1.8.5 should generate a violation
```
Comment (chalasr) [inline tests/Requirement/RequirementCheckerTest.php:155]: Needs a new `RequirementViolationList()` to make tests green

File: tests/Requirement/RequirementCheckerTest.php [hunk 2]
```diff
@@ -135,39 +135,49 @@ public function testCheckerShouldNotCreateAViolationIfBinaryIsOK()
...
+        $this->processor->process('vagrant')->willReturn('vagrant 1.8.7');
         $this->requirementChecker->check($requirement, $violationList);
+        var_dump($violationList);
         $this->assertEmpty($violationList);
 
         // Version 1.9.* should pass
         $violationList = new RequirementViolationList();
         $this->processor->process('vagrant')->willReturn('vagrant 1.9.2');
         $this->requirementChecker->check($requirement, $violationList);
+        var_dump($violationList);
```
Comment (chalasr) [inline tests/Requirement/RequirementCheckerTest.php:180]: to revert :)

File: src/Requirement/RequirementRepository.php [hunk 3]
```diff
@@ -24,16 +24,16 @@ public static function getRequirements(): array
                 'vagrant',
                 RequirementType::BINARY(),
                 RequirementLevel::REQUIRED(),
-                '^1.8.4',
-                'Darwin' === php_uname('s') ? ['1.8.5.*', '1.8.6.*', '1.8.7.*'] : ['1.8.5.*', '1.8.6.*'], // /!\ Exclude 1.8.7 for OSX since it is buggy
```
Comment (supertanuki) [inline src/Requirement/RequirementRepository.php:28]: @chalasr is it ok to remove this line?

**Comment by supertanuki** [comment]
@chalasr I close this PR ?

**Comment by chalasr** [comment]
@supertanuki I'm going to test the following setup: ` vagrant 2.0.1+, Virtualbox 5.2, ansible 2.4.2+` (see #137), will update this PR accordingly right after, please don't close :)

**Pull request closed without merging**